### PR TITLE
Org:Sandbox Status

### PR DESCRIPTION
**Pull request by maggiben**
### What does this PR do ?
Adds a `sandboxStatus` method to the `Org` class

WI Reference:
@W-10657353@

### CODE REVIEW

File: src/org.ts [hunk 1]
```diff
@@ -640,6 +654,69 @@ export class Org extends AsyncCreatable<Org.Options> {
...
+  }
+
+  /**
+   * Polls the sandbox org for the sandboxProcessObject.
+   *
+   * @param sandboxProcessObj: The in-progress sandbox signup request
+   * @param options { wait?: Duration; interval?: Duration }
+   * @returns {SandboxProcessObject}
+   */
+  private async authWithRetries(
```
Comment (WillieRuemmele) [inline src/org.ts:704]: can we remove these three private methods and move it all under the public `sandboxStatus`?

File: src/org.ts [hunk 2]
```diff
@@ -640,6 +654,69 @@ export class Org extends AsyncCreatable<Org.Options> {
...
+  /**
+   * Query the sandbox for the SandboxProcessObject by sandbox name
+   *
+   * @param sandboxName The name of the sandbox to query
+   * @returns {SandboxProcessObject} The SandboxProcessObject for the sandbox
+   */
+  private async queryLatestSandboxProcessBySandboxName(sandboxNameIn: string): Promise<SandboxProcessObject> {
+    const { tooling } = this.getConnection();
+    const SOBJECT_SANDBOXPROCESS = 'SandboxProcess';
+    const QUERY_SANDBOXPROCESS_FIELDS =
```
Comment (WillieRuemmele) [inline src/org.ts:666]: can these vars be injected into `queryStr` directly? They aren't referenced anywhere else

File: src/org.ts [hunk 3]
```diff
@@ -640,6 +654,69 @@ export class Org extends AsyncCreatable<Org.Options> {
...
+   * @param options { wait?: Duration; interval?: Duration }
+   * @returns {SandboxProcessObject}
+   */
+  private async authWithRetries(
+    sandboxProcessObj: SandboxProcessObject,
+    options: { wait?: Duration; interval?: Duration }
+  ): Promise<SandboxProcessObject> {
+    const retries = options.wait ? options.wait.seconds / Duration.seconds(30).seconds : 0;
+    const pollInterval = options.interval ?? Duration.seconds(30);
+    // const maxPollingRetries = this.getMaxPollingRetries();
```
Comment (WillieRuemmele) [inline src/org.ts:710]: a leftover comment

**Review by WillieRuemmele — APPROVED** [review]

**Review by shetzel — APPROVED** [review]
Approved with nits.

**Pull request merged**